### ISSUE TITLE
Support globally installed custom rules

### ISSUE THREAD

**Issue by MatisseHack** [DavidAnson/markdownlint-cli2#24]
Is there any interest in allowing globally installed modules to be referenced as `customRules` in a JSON configuration file (without using an absolute path)?

For example:

```shell
npm install -g some-custom-rule
```

Could be referenced by:

```json
{
  "customRules": ["some-custom-rule"]
}
```

**Comment by DavidAnson** [comment]
This may/should work today based on `require` behavior resolution paths. (There's already a test that verifies it works if CLI2 and the rule are both globally installed: https://github.com/DavidAnson/markdownlint-cli2/runs/3143722003?check_suite_focus=true)

Have you tried?

**Comment by MatisseHack** [comment]
Oh, my mistake! I assumed it was not working because the module was installed globally, but the issue was with my configuration file! I was trying to use a `.markdownlint-cli2.json` file instead of a `.markdownlint-cli2.jsonc` file. Would there be any interest in allowing the regular `.json` extension for unobservant people like me 😂 ?

Unfortunately, even though I have `markdownlint-cli2` working with a global module, I'm now having this exact same issue with the VS Code extension. VS Code will only load a custom rule if it is installed in the workspace or if I specify the absolute path to the global module. Is this a known limitation or should I create a new issue over in that repo?

**Comment by DavidAnson** [comment]
There are already eight different configuration file names/extensions that are supported, so I am pretty reluctant to add another. :)

It seems we can close this issue. It would be reasonable to open a similar one in the extension repository, but I'm not sure whether I'd decide it was appropriate to "fix". The issue in my mind right now is that VS Code doesn't really have a notion of "global packages" like the Node environment does, so I'm not sure that what you're trying to do technically makes sense in there (although I understand why it makes sense as a person).

**Comment by MatisseHack** [comment]
Haha, there are quite a few formats supported already! I guess I saw `.json` and `.jsonc` as the same format and didn't think the extension would make much of a difference (the same way I might expect to use `.yml` and `.yaml` interchangeably). Not a huge deal though once I started looking in the right place for my issue.

Ah, I see! I created DavidAnson/vscode-markdownlint#180 to continue the conversation.
